build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/dependencies/patches/Utils.java

```diff
@@ -101,9 +109,15 @@ public static void patchJar(File inputFile, File outputFile, Collection<PatcherI
                         );
                     }
                 } else {
-                    // Read the entry's data and write it to the new JAR
-                    try (InputStream is = jarFile.getInputStream(entry)) {
-                        is.transferTo(jos);
+                    if (unsignJar && entryName.equals("META-INF/MANIFEST.MF")) {
+                        var manifest = new Manifest(jarFile.getInputStream(entry));
+                        manifest.getEntries().clear();
+                        manifest.write(jos);
```

Comment:
this means we keep the "main" attributes, e.g.

```
Manifest-Version: 1.0
Implementation-Title: Microsoft Azure Java Core Library
Implementation-Version: 1.55.3
Build-Jdk-Spec: 21
Created-By: Maven JAR Plugin 3.4.2
Implementation-Vendor: Microsoft Corporation
```

Reply:
Just an idea; instead of a fixed "unsign", why don't you add a manifest transformer?
e.g. 
```
public static void patchJar(..., ) {
....
   if (entryName.equals("META-INF/MANIFEST.MF")) {
      var manifest = new Manifest(jarFile.getInputStream(entry));
      manifestConsumer.accept(manifest);
      manifest.write(jos);
   }
```
This way the logic can be more precise and cover more scenarios